xbmc/epg/EpgContainer.cpp

```diff
@@ -245,16 +245,17 @@ bool CEpgContainer::PersistAll(void)
 {
   bool bReturn(true);
   m_critSection.lock();
-  std::map<unsigned int, CEpg*> copy = m_epgs;
+  EPGMAP copy = m_epgs;
   m_critSection.unlock();
 
-  for (EPGMAP_CITR it = copy.begin(); it != copy.end() && !m_bStop; it++)
+  for (const auto &epgEntry : copy)
   {
-    CEpg *epg = it->second;
+    CEpg *epg = epgEntry.second;
     if (epg && epg->NeedsSave())
-    {
       bReturn &= epg->Persist();
-    }
+
+    if (m_bStop)
```

Comment:
> because it is extremely easy to miss the && !m_bStop in the for loop definition. The new code is much more obvious to read for me.

I had the same discussion with @opdenkamp before and I prefer the break condition to be inside the loop, which improves readability a lot. But as said it's a matter of taste.